src/java/main/org/apache/zookeeper/server/persistence/FileTxnLog.java

```diff
@@ -399,18 +403,20 @@ public boolean truncate(long zxid) throws IOException {
             }
             long pos = input.getPosition();
             // now, truncate at the current position
-            RandomAccessFile raf=new RandomAccessFile(itr.logFile,"rw");
+            raf = new RandomAccessFile(itr.logFile, "rw");
             raf.setLength(pos);
-            raf.close();
             while(itr.goToNextLog()) {
-                if (!itr.logFile.delete()) {
-                    LOG.warn("Unable to truncate {}", itr.logFile);
+                try {
+                    Files.delete(itr.logFile.toPath());
+                } catch (NoSuchFileException e) {
                 }
             }
         } finally {
             close(itr);
+            if (raf != null) {
+                raf.close();
+            }
         }
-        return true;
     }
```

Comment:
IMHO.should we put code block `Line416-Line418 ` before ` Line415` ?

Reply:
o, sorry. I can't realize what the difference both.  Can you make me clear about that, then  I can fix it better?  Thanks